src/VkBootstrap.h

```diff
@@ -849,11 +851,14 @@ class SwapchainBuilder {
 	SwapchainBuilder& use_default_image_usage_flags();
 
 	// Set the bitmask of the format feature flag for acquired swapchain images.
+	// If the desired formats cannot allow it, building the swapchain will result in the
+	// `SwapchainError::required_usage_or_features_not_supported` error. Use this functionnality only if you require
+	// formats to support a set of feature flags that are not already implied by set_image_usage_flags
 	SwapchainBuilder& set_format_feature_flags(VkFormatFeatureFlags feature_flags);
 	// Add a format feature to the bitmask for acquired swapchain images.
 	SwapchainBuilder& add_format_feature_flags(VkFormatFeatureFlags feature_flags);
 	// Use the default format feature bitmask values. This is the default if no format features
-	// are provided. The default is VK_FORMAT_FEATURE_SAMPLED_IMAGE_BIT
+	// are provided. The default is 0
 	SwapchainBuilder& use_default_format_feature_flags();
```

Comment:
Why keep the ability to set format feature flags? My understanding is that format feature flags are something the application queries from the driver about a format, and the usage flags are used to tell the driver what they want to do with an image. To me, that would mean we only need to keep usage flags around for the user to do. Definitely let me know if I am mistaken about this.

Reply:
First, I backtracked a bit because I didn't want to remove a function that some people may use.
Then, I wasn't sure if anyone relied on this (if there is any use-case where one has to use format features instead of image usages). 

My understanding is the same as yours. Furthermore, the mapping `image usage` --> `format feature flags` seems to be 1 --> N, meaning that if the implementation reports a supported usage in `vkGetPhysicalDeviceSurfaceCapabilitiesKHR`, it also supports all associated format features. Thus it would be irrelevant to use the format feature flags

So for the very unique case where the driver reports `supportedUsageFlags` in `vkGetPhysicalDeviceSurfaceCapabilitiesKHR`, instead of the usual feature flags, we shouldn't need to query feature flags anymore. It is still unclear to me why they report supportedUsages instead of FeatureFlags.

Since Monday, I convinced myself that this API is not useful, so I will remove it in next commit.

Reply:
Great, the removal makes the interface just that bit smaller and thus simpler.

Plus, while vk-bootstrap tries to validate what it can, it can only do so much within reason. Better to let validation layers do the heavy lifting.